sdk/servicebus/service-bus/test/internal/2048-limit-test.spec.ts

```diff
@@ -0,0 +1,311 @@
+import { checkWithTimeout, TestClientType, TestMessage } from "../public/utils/testUtils";
+import { ServiceBusReceiver } from "../../src/receivers/receiver";
+import {
+  ServiceBusClientForTests,
+  createServiceBusClientForTests,
+  EntityName
+} from "../public/utils/testutils2";
+import { verifyMessageCount } from "../public/utils/managementUtils";
+import { delay } from "rhea-promise";
+
+chai.use(chaiAsPromised);
+
+let serviceBusClient: ServiceBusClientForTests;
+let entityName: EntityName;
+let sender: ServiceBusSender;
+let receiver: ServiceBusReceiver;
+const numberOfMessagesToSend = 3000;
+
+const testClientTypes = [
+  TestClientType.PartitionedQueue,
+  TestClientType.PartitionedQueueWithSessions,
+  TestClientType.UnpartitionedQueue,
+  TestClientType.UnpartitionedQueueWithSessions
+];
+
+async function beforeEachTest(
+  entityType: TestClientType,
+  receiveMode: "peekLock" | "receiveAndDelete" = "peekLock"
+): Promise<void> {
+  entityName = await serviceBusClient.test.createTestEntities(entityType);
+  if (receiveMode === "receiveAndDelete") {
+    receiver = await serviceBusClient.test.createReceiveAndDeleteReceiver(entityName);
+  } else {
+    receiver = await serviceBusClient.test.createPeekLockReceiver(entityName, {
+      // maxAutoLockRenewalDurationInMs: 0
+    });
+  }
+
+  sender = serviceBusClient.test.addToCleanup(
+    serviceBusClient.createSender(entityName.queue ?? entityName.topic!)
+  );
+}
+
+function afterEachTest(): Promise<void> {
+  return serviceBusClient.test.afterEach();
+}
+
+async function sendMessages(numberOfMessagesToSend: number) {
+  let current = 0;
+  const messageBodies = [];
+  while (current < numberOfMessagesToSend) {
+    const batch = await sender.createMessageBatch();
+    let body = `message-${current}`;
+    while (
+      current < numberOfMessagesToSend &&
+      batch.tryAddMessage(
+        entityName.usesSessions ? { body, sessionId: TestMessage.sessionId } : { body }
+      )
+    ) {
+      messageBodies.push(body);
+      current++;
+      body = `message-${current}`;
+    }
+
+    await sender.sendMessages(batch);
+  }
+  return messageBodies;
+}
+
+describe("2048 scenarios - receiveBatch in a loop", function(): void {
+  before(() => {
+    serviceBusClient = createServiceBusClientForTests({
+      retryOptions: { maxRetries: 1, retryDelayInMs: 10 }
+    });
+  });
+
+  after(() => {
+    return serviceBusClient.test.after();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function receiveMessages(numberOfMessagesToReceive: number) {
+    let messages: ServiceBusReceivedMessage[] = [];
+    while (messages.length < numberOfMessagesToReceive) {
+      messages = messages.concat(await receiver.receiveMessages(50));
+    }
+    chai.assert.equal(
+      messages.length,
+      numberOfMessagesToReceive,
+      "Unexpected number of messages received"
+    );
+    return messages;
+  }
+
+  describe("receiveAndDelete", () => {
```

Comment:
log an issue for/with the message loss repro

Reply:
Logged https://github.com/Azure/azure-sdk-for-js/issues/15115 - (current PR is unrelated to the bug).